### ISSUE TITLE
Parse error on "update.php" at installation of Semantic MediaWiki

### ISSUE THREAD

**Issue by henrigilabert** [SemanticMediaWiki/SemanticMediaWiki#4148]
### Setup and configuration

- SMW version: 3.0.2
- MW version:1.33.0
- PHP version:7.1.30
- DB system (MySQL, Blazegraph, etc.) and version: MySQL 5.6.40-84

### Issue
After installing SMW correctly with composer, I run update.php form the manitenance directory and I get the message : ```Parse error: syntax error, unexpected '?' in /home/.../maintenance/update.php on line 254```

**Comment by mwjames** [comment]
> ### Issue
> after installing SMW correctly with composer, I run update.php form the
> manitenance directory and I get the message : Parse error: syntax error,
> unexpected '?' in /home/.../maintenance/update.php on line 254

Well, if you look at the mentioned line you'll find `??` and as far as
I remember that is an issue in MediaWiki that relates to the used
syntax.

For details, see [0, 1] where it has been noted "... A user reported
in #mediawiki today with this exact situation (using php 7.1 but his
webhoster had php cli symlinked to to php56)."

@kghbln Given that users are hitting this issue, I think it is time to
see to it that this is backported soonish and have a MW 1.33.1
release.

[0] https://gerrit.wikimedia.org/r/#/c/mediawiki/core/+/522095/
[1] https://phabricator.wikimedia.org/T213893#5320412

**Comment by kghbln** [comment]
I have created a back-port of one of the commits. Let's see what happens. Anyways I am pretty sure that we will not see a new MW release soon. This is just the way it is.

**Comment by henrigilabert** [comment]
Thank you for your help, I've applied corrections suggested, the script update.php lokks ok... but now I have Parse error: syntax error, unexpected '?' in /home/.../maintenance/Maintenance.php on line 882.
the line of code in the script is : $this->setParam( $options, $bits[0], $bits[1] ?? 1 );

**Comment by kghbln** [comment]
@henrigilabert I believe your hoster has to remove PHP 5.6 from the system. The change will only give you a graceful failure the issue will persist no matter what you do.

**Comment by henrigilabert** [comment]
According to mediawiki info the PHP version is 7.1.30

Cordialement, - Best regards - Mit freundlichen Grüßen - Met vriedelijke groeten - Saludos - Salut - Hälsningar - Ciao

Tél. 06.75.23.84.11 email henri.gilabert@sla.lu<mailto:henri.gilabert@sla.lu>

Henri GILABERT

P  Pensez à l’environnement avant d’imprimer / Before printing, think about the environment.

De : Karsten Hoffmeyer <notifications@github.com>
Envoyé : mardi 16 juillet 2019 15:02
À : SemanticMediaWiki/SemanticMediaWiki <SemanticMediaWiki@noreply.github.com>
Cc : Henri GILABERT <henri.gilabert@sla.lu>; Mention <mention@noreply.github.com>
Objet : Re: [SemanticMediaWiki/SemanticMediaWiki] Parse error on "update.php" at installation of Semantic MediaWiki (#4148)


@henrigilabert<https://github.com/henrigilabert> I believe your hoster has to remove PHP 5.6 from the system. The change will only give you a graceful failure the issue will persist no matter what you do.

**Comment by kghbln** [comment]
I am not sure if you read the linked information.

**Comment by henrigilabert** [comment]
I did and I double checked the PHP version of the hoster, see below

[cid:image001.jpg@01D53BE8.FCF788F0]

Cordialement, - Best regards - Mit freundlichen Grüßen - Met vriedelijke groeten - Saludos - Salut - Hälsningar - Ciao

Tél. 06.75.23.84.11 email henri.gilabert@sla.lu<mailto:henri.gilabert@sla.lu>

Henri GILABERT

P  Pensez à l’environnement avant d’imprimer / Before printing, think about the environment.

De : Karsten Hoffmeyer <notifications@github.com>
Envoyé : mardi 16 juillet 2019 15:07
À : SemanticMediaWiki/SemanticMediaWiki <SemanticMediaWiki@noreply.github.com>
Cc : Henri GILABERT <henri.gilabert@sla.lu>; Mention <mention@noreply.github.com>
Objet : Re: [SemanticMediaWiki/SemanticMediaWiki] Parse error on "update.php" at installation of Semantic MediaWiki (#4148)


I am not sure if you read the linked information.

**Comment by kghbln** [comment]
I created a help page [for this](https://www.semantic-mediawiki.org/wiki/Help:Installation/Troubleshooting/Parse_error:_syntax_error,_unexpected)

**Comment by mwjames** [comment]
> update.php lokks ok... but now I have Parse error: syntax error, unexpected
> '?' in /home/.../maintenance/Maintenance.php on line 882.
> the line of code in the script is : $this->setParam( $options, $bits[0],
> $bits[1] ?? 1 );

Again, this is a syntax related issue because MW 1.33 requires "...
requires PHP 7.0.13 or later." [0] but somehow your PHP linked library
uses anything but PHP 7 even though `Special:Version` shows 7.1*.

Whether you have SMW installed or not, this issue will remain as long
as your hosting provider is linking to an unsupported PHP.

Details about  the `??` (Null coalescing operator) can be found at [1].

[0] https://github.com/wikimedia/mediawiki/blob/REL1_33/RELEASE-NOTES-1.33#L485-L488
[1] https://www.php.net/manual/en/migration70.new-features.php

**Comment by henrigilabert** [comment]
After reading your help page I’ve checked the command line php version and… it is version 5.6.40

There is nothing I can do ?

Cordialement, - Best regards - Mit freundlichen Grüßen - Met vriedelijke groeten - Saludos - Salut - Hälsningar - Ciao

Tél. 06.75.23.84.11 email henri.gilabert@sla.lu<mailto:henri.gilabert@sla.lu>

Henri GILABERT

P  Pensez à l’environnement avant d’imprimer / Before printing, think about the environment.

De : Karsten Hoffmeyer <notifications@github.com>
Envoyé : mardi 16 juillet 2019 15:17
À : SemanticMediaWiki/SemanticMediaWiki <SemanticMediaWiki@noreply.github.com>
Cc : Henri GILABERT <henri.gilabert@sla.lu>; Mention <mention@noreply.github.com>
Objet : Re: [SemanticMediaWiki/SemanticMediaWiki] Parse error on "update.php" at installation of Semantic MediaWiki (#4148)


I created a help page for this<https://www.semantic-mediawiki.org/wiki/Help:Installation/Troubleshooting/Parse_error:_syntax_error,_unexpected>

**Comment by kghbln** [comment]
> There is nothing I can do ?

What about contacting your hoster about this? I have already suggested this.

**Comment by henrigilabert** [comment]
I got an answer from the hoster : the command line has to pb php71 to have it running php 7.1
Now the issue is fixed but I have an other problem from update.php telling me that my composer.lock is not up to date

Cordialement, - Best regards - Mit freundlichen Grüßen - Met vriedelijke groeten - Saludos - Salut - Hälsningar - Ciao

Tél. 06.75.23.84.11 email henri.gilabert@sla.lu<mailto:henri.gilabert@sla.lu>

Henri GILABERT

P  Pensez à l’environnement avant d’imprimer / Before printing, think about the environment.

De : Karsten Hoffmeyer <notifications@github.com>
Envoyé : mardi 16 juillet 2019 15:26
À : SemanticMediaWiki/SemanticMediaWiki <SemanticMediaWiki@noreply.github.com>
Cc : Henri GILABERT <henri.gilabert@sla.lu>; Mention <mention@noreply.github.com>
Objet : Re: [SemanticMediaWiki/SemanticMediaWiki] Parse error on "update.php" at installation of Semantic MediaWiki (#4148)


There is nothing I can do ?

What about contacting your hoster about this? I have already suggested this.
